### PR TITLE
feat(download): device filename template engine (#208)

### PR DESCRIPTION
**Pull request by lqdev**
## Summary

Closes #208 — adds a pure, exhaustively-tested template engine for rendering
device-side filenames. Engine only; the sync wiring that consumes it lands in #209.

## Changes

- **New** `src/download/device_template.rs` — `render()`, `disambiguate()`, `TemplateError`, `DeviceFilenameOptions`. No I/O, no `tokio`. 23 unit tests cover each token, error paths, length caps, and edge cases.
- **New** `src/download/sanitize.rs` — extracted the comprehensive cross-platform sanitizer (Windows reserved names, Unicode folding, length cap) out of `download::manager` so both manager and the template engine share one implementation. 5 unit tests.
- **`src/download/manager.rs`** — removed the duplicated `sanitize_filename` / `handle_reserved_names` methods; the two call sites now invoke the shared free functions.
- **`src/download/mod.rs`** — registers `device_template` and `sanitize` submodules.
- **`CHANGELOG.md`** — entry under `[Unreleased]` ### Added.

## Token reference

| Token | Substitution |
|-------|--------------|
| `{podcast}` | Podcast title, sanitized |
| `{podcast_short}` | Podcast title, truncated to 30 chars |
| `{title}` | Episode title, sanitized |
| `{track}` | Episode number (no padding); empty if missing |
| `{track:NN}` | Episode number, zero-padded (e.g. `{track:03}` → `007`) |
| `{date}` | Published date, default `YYYY-MM-DD` |
| `{date:%fmt}` | Published date with chrono strftime format |
| `{ext}` | File extension (without leading dot) |

Literal `/` in the template becomes a path separator; `/` inside a substituted token value is sanitized into `-` so a podcast titled `Foo/Bar` cannot escape its segment.

## Acceptance criteria

- [x] `cargo test` passes (23 new template tests + 5 new sanitizer tests + all existing 638)
- [x] Pure module — no `tokio`, no `std::fs` imports
- [x] `sanitize_filename` exists in exactly ONE place after the refactor
- [x] No new public dependencies (uses existing `chrono`, `thiserror`)
- [x] Module-level rustdoc with token table and example

## Manual Testing

N/A — pure engine, not yet wired into `sync_to_device`. The wiring (and observable user-visible behavior) lands in #209. All behavior is covered by the 28 unit tests across the two new modules.

## Quality

- `cargo fmt` ✅
- `cargo clippy --all-targets -- -D warnings` ✅
- `cargo test` ✅ (638 tests pass)

### CODE REVIEW

File: src/download/device_template.rs [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+    let (name, param) = match token.find(':') {
+        Some(idx) => (&token[..idx], Some(&token[idx + 1..])),
+        None => (token, None),
+    };
+
+    match (name, param) {
+        ("podcast", None) => Ok(podcast.title.clone()),
+        ("podcast_short", None) => Ok(truncate_chars(&podcast.title, PODCAST_SHORT_MAX)),
+        ("title", None) => Ok(episode.title.clone()),
+        ("track", None) => Ok(episode
```
[the remark this answers is not in the document]
Comment (lqdev) [reply src/download/device_template.rs]: Accepted ΓÇö fixed in 6dc24a1. `{podcast_short}` now sanitizes the raw title before truncating to 30 chars, matching the documented behaviour. New test `token_podcast_short_truncates_after_sanitization` covers the `&` ΓåÆ `and` expansion case.

File: src/download/sanitize.rs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+//! * Latin-1 / Latin Extended accented characters → ASCII equivalents
+//! * Misc symbols (`&` → `and`, `@` → `at`, etc.)
+//! * Windows reserved device names (`CON`, `PRN`, `AUX`, `NUL`, `COM1-9`, `LPT1-9`)
+//! * Length cap of 140 characters with UTF-8 boundary safety
+
+/// Comprehensive cross-platform filename sanitization.
+///
+/// `is_folder` only affects the fallback name when sanitization yields an
+/// empty string (`"Podcast"` for folders, `"Episode"` otherwise).
+pub fn sanitize_filename(input: &str, is_folder: bool) -> String {
```
[the remark this answers is not in the document]
Comment (lqdev) [reply src/download/sanitize.rs:26]: Accepted ΓÇö doc updated in 6dc24a1 to match actual behaviour: empty/whitespace input always returns `Untitled` regardless of `is_folder`; `is_folder` only chooses the fallback (`Podcast`/`Episode`) when a non-empty input sanitizes to empty. Preserves backward-compat with the lifted `manager.rs` implementation.

**Comment by lqdev** [comment]
Copilot review actioned in 6dc24a1.

**Accepted (5):**
- `{podcast_short}` sanitize-then-truncate ordering
- `cap_bytes(0)` returns empty + render() falls back to `_` placeholder so we never emit a `.mp3` dotfile
- `{track:0}` rejected as Malformed
- `{episode_number}` / `{episode_number:NN}` aliases for `{track}` (keeps DeviceProfile rustdoc from #207 working against the engine)
- `sanitize_filename` rustdoc clarified to match actual `is_folder` behaviour

Added 4 new unit tests covering the above; all 638+ tests and `clippy --all-targets -D warnings` pass.

**Deferred (2):**
- 140-byte filename cap should live in `src/constants.rs` ΓåÆ #217
- Two `sanitize_filename` functions (`download::sanitize` vs `utils::validation`) need consolidation ΓåÆ #218

Both are valid concerns but pre-existing and out of scope for this PR (which is the device template engine, not a constants/sanitization audit).
